src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs

```diff
@@ -215,14 +223,37 @@ private void InitializeDataCollectors(string runSettings, ITestEventsPublisher t
             {
                 var outOfProcDataCollectionManager = new ProxyOutOfProcDataCollectionManager(DataCollectionTestCaseEventSender.Instance, testEventsPublisher);
             }
-             
+
             // Initialize inproc data collectors if declared in run settings.
             if (XmlRunSettingsUtilities.IsInProcDataCollectionEnabled(runSettings))
             {
                 var inProcDataCollectionExtensionManager = new InProcDataCollectionExtensionManager(runSettings, testEventsPublisher, defaultCodeBase, TestPluginCache.Instance);
             }
         }
 
+        private void TestSessionMessageHandler(object sender, TestRunMessageEventArgs e)
+        {
+            if (this.testMessageEventsHandler != null)
+            {
+                if (e.Level == TestMessageLevel.Error)
```

Comment:
same as discovery handler

Reply:
Removed Level downgrade lines, and now it follows whatever the Level it gets.